Trying out the K nearest neighbors regression model

In [3]:
from mglearn import datasets
X_waves, y_waves = datasets.make_wave(n_samples=40)

In [4]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X_waves, y_waves, random_state=0)

In [13]:
from sklearn.neighbors import KNeighborsRegressor
knnr = KNeighborsRegressor(n_neighbors=3)
knnr.fit(X_train, y_train)
score = knnr.score(X_test, y_test)

print(f"score: {score:.2f}")

score: 0.83


The coefficient of determination is 0.83, which isn't bad honestly.